mpp/shmem_c_func.m4

```diff
@@ -5,105 +5,114 @@ dnl This file is part of the Sandia OpenSHMEM software package. For license
 dnl information, see the LICENSE file in the top level directory of the
 dnl distribution.
 dnl
+
+#ifndef SHMEM_FUNCTION_ATTRIBUTES
+#  if OPAL_C_HAVE_VISIBILITY == 1
+#     define SHMEM_FUNCTION_ATTRIBUTES __attribute__((visibility("default")))
+#  else
+#     define SHMEM_FUNCTION_ATTRIBUTES
+#  endif
+#endif
```

Comment:
This will go into the public header.  This means that we would also cause user code to make these symbols visible.  Do we need to include the attributes there?